vendor/assets/javascripts/behaviors/selectable_asset.js

```diff
@@ -10,6 +10,9 @@ Cmsify.SelectableAsset.prototype.select = function() {
   $preview.html(this.renderAssetOriginalPreview());
   $preview.append($('<p />').html(this.getFileName()));
   $('.js-remove-asset').removeClass('uk-hidden');
+  if (this.$el.data('asset-name') === 'download') {
+    $('#resource_item_name').val(this.getFileName());
+  }
```

Comment:
So ran into a problem with this. When you don't have the check it will change the name field when uploading or changing the featured image.